lib/properties.js

```diff
@@ -13,8 +14,8 @@ class Properties extends EventEmitter {
 
     this.initialized = false;
 
-    this.layers = [];
-    this.properties = {};
+    this._layers = Immutable.List();
+    this._properties = Immutable.Map();
```

Comment:
`Immutable` is going to be introduced in #188 anyway. Not sure why we shouldn't take advantage of it.